Neos.Neos/Classes/Command/SiteCommandController.php

```diff
@@ -111,31 +156,93 @@ public function createCommand($name, $packageKey, $nodeType, $nodeName = null, $
     }
 
     /**
-     * Remove site with content and related data (with globbing)
+     * Import sites
+     *
+     * This command allows importing sites from the given path/package. The format must
+     * be identical to that produced by the exportAll command.
+     *
+     * !!! The live workspace has to be empty prior to importing. !!!
+     *
+     * If a path is specified, this command expects the corresponding directory to contain the exported files
      *
-     * In the future we need some more sophisticated cleanup.
+     * If a package key is specified, this command expects the export files to be located in the private resources
+     * directory of the given package (Resources/Private/Content).
      *
-     * @param string $siteNode Name for site root nodes to clear only content of this sites (globbing is supported)
+     * @param string|null $packageKey Package key specifying the package containing the sites content
+     * @param string|null $path relative or absolute path and filename to the export files
      * @return void
      */
-    public function pruneCommand($siteNode)
+    public function importAllCommand(string $packageKey = null, string $path = null, string $contentRepository = 'default', bool $verbose = false): void
     {
-        $sites = $this->findSitesByNodeNamePattern($siteNode);
-        if (empty($sites)) {
-            $this->outputLine('<error>No Site found for pattern "%s".</error>', [$siteNode]);
-            // Help the user a little about what he needs to provide as a parameter here
-            $this->outputLine('To find out which sites you have, use the <b>site:list</b> command.');
-            $this->outputLine('The site:prune command expects the "Node name" from the site list as a parameter.');
-            $this->outputLine('If you want to delete all sites, you can run <b>site:prune \'*\'</b>.');
-            $this->quit(1);
+        // Since this command uses a lot of memory when large sites are imported, we warn the user to watch for
+        // the confirmation of a successful import.
+        $this->outputLine('<b>This command can use a lot of memory when importing sites with many resources.</b>');
+        $this->outputLine('If the import is successful, you will see a message saying "Import of site ... finished".');
+        $this->outputLine('If you do not see this message, the import failed, most likely due to insufficient memory.');
+        $this->outputLine('Increase the <b>memory_limit</b> configuration parameter of your php CLI to attempt to fix this.');
+        $this->outputLine('Starting import...');
+        $this->outputLine('---');
```

Comment:
ha this is copied from Neos 8.3 :D but im not sure if this is still true? In 8.3 the xml parsing was super expensive but now we just read json line by line and see ...